library/Network/CloudSeeder/CommandLine.hs

```diff
@@ -107,21 +107,23 @@ provision phase = subparser . command "provision" $ info parser infoMod
         helper' p = (abortOption ShowHelpText opts <*> empty) <|> p
           where opts = long "help" <> short 'h' <> internal
 
-    optionsParser :: S.Set ParameterSpec -> Parser (M.Map T.Text T.Text)
+    optionsParser :: S.Set ParameterSpec -> Parser (M.Map T.Text (Maybe T.Text))
     optionsParser specs = M.fromList <$> traverse parameter (S.toList specs)
       where
         parameter spec = do
           let key = spec ^. parameterKey
               keyStr = T.unpack key
-          val <- textOption (long keyStr <> metavar keyStr <> defaultMod spec)
+          val <- defaultOption $ long keyStr <> metavar keyStr <> defaultMod spec
           pure (key, val)
 
-        defaultMod (Required _) = mempty
-        defaultMod (Optional _ defVal) = value (T.unpack defVal)
+        defaultMod (Optional _ x) = value x
+        defaultMod _ = mempty
```

Comment:
On second glance, you could probably replace this entire function with `(^. _Optional._2.to value)` because modifiers are monoidal. Is that better? Honestly, probably not. But it’s a fun trick.